static/js/tools/map/app.tsx

```diff
@@ -100,7 +100,9 @@ function applyHash(context: ContextType): void {
 function updateHash(context: ContextType): void {
   let hash = updateHashStatVarInfo("", context.statVarInfo.value);
   hash = updateHashPlaceInfo(hash, context.placeInfo.value);
-  if (hash) {
-    history.pushState({}, "", `/tools/map#${encodeURIComponent(hash)}`);
+  const newHash = encodeURIComponent(hash);
+  const currentHash = location.hash.replace("#", "");
+  if (newHash && newHash !== currentHash) {
+    history.pushState({}, "", `/tools/map#${newHash}`);
```

Comment:
cool :) though, consider adding the URI path as a constant.

Reply:
done